### ISSUE TITLE
Generic Accept header value */* and Service Workers

### ISSUE THREAD

**Issue by tomayac** [whatwg/fetch#634]
Generic ```Accept``` headers like, for example, ```*/*``` for [images in Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1249474) make dealing with such requests in a Service Worker context [hard](http://twitter.com/justmarkup/status/926793878950367232) and encourage relying on [file extension sniffing](https://justmarkup.com/log/2017/11/network-based-image-loading/), especially as there is [no consistency](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation/List_of_default_Accept_values#Values_for_an_image) among how browsers deal with setting this value.

This issue is somewhat related to https://github.com/whatwg/fetch/issues/521 for dynamically created ```Request```s , but also covers static fetches coming from ```img```, ```script```, ```link```, *etc.*

@wanderview [suggested](https://github.com/w3c/ServiceWorker/issues/1206#issuecomment-341991783) to open an Issue with Fetch in order to discuss this.

**Comment by annevk** [comment]
Currently https://fetch.spec.whatwg.org/#concept-fetch recommends a particular `Accept` header for image requests. If browsers don't have that it would be a bug. (As is not supporting `destination`.)

**Comment by wanderview** [comment]
Note, we send */* for image requests, but not we get some other accept headers wrong.  I have this bug open to align with the spec:

https://bugzilla.mozilla.org/show_bug.cgi?id=1417463

**Comment by tomayac** [comment]
I guess ```request.destination``` would solve this, iff ("ff") universally available. For the ```Accept``` header discussion, I found Firefox's https://bugzilla.mozilla.org/show_bug.cgi?id=1249474 quite enlightening, where the move to ```*/*``` was motivated. Not sure where this leaves us (and as I am new here, I definitely missed a ton of the discussions that were had, so sorry for any comment that for the people longer around may seem dumb).

**Comment by annevk** [comment]
@tomayac please don't worry about asking questions. They're essential. (If we keep getting certain questions a lot, that's probably a reason to start writing some documentation.)

I think the main takeaway here is that browsers need to fix various issues. An active thing that you or someone else could take up is to ensure there is web-platform-tests test coverage and appropriate browser bugs are filed (and maybe linked from this issue for completeness). I don't think anything here warrants a change to the standard right now.

**Comment by tomayac** [comment]
Thanks for the kind words, @annevk! 

Looking for bugs to track, so far I found two:

- Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1402892
- Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=520460

There doesn't seem to be anything yet for [Edge](https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/?page=1&q=request.destination) or [WebKit](https://bugs.webkit.org/buglist.cgi?quicksearch=request.destination), but I might just not be digging deeply enough. Anyone aware of bugs there, or for other browsers?

**Comment by wanderview** [comment]
FWIW, I think destination implementation has been slow for a couple reasons:

1. We had not heard developers asking for it yet with use cases.  So this feedback is good.
2. The spec changed a lot from the initial RequestContext spec, so the feature got put on the back burner while that all happened.  It seems pretty stable now, though.
3. There aren't really any WPT tests for it yet which makes first implementation more work.  It means whoever does this first will probably have more work to not only write the test, but ensure the spec works as intended.

Anyway, I'll see if we can bump up the priority of the firefox implementation.  It probably won't be immediate, but maybe we can do it in Q1 of 2018.

**Comment by tomayac** [comment]
Really happy about Safari Technology Preview's implementation of `request.destination`:

![screen shot 2017-12-21 at 10 28 30](https://user-images.githubusercontent.com/145676/34249402-eebe2448-e639-11e7-8bff-16062f11d926.png)
![screen shot 2017-12-21 at 10 28 21](https://user-images.githubusercontent.com/145676/34249403-eed932ce-e639-11e7-987b-442b0b1cbbcc.png)
![screen shot 2017-12-21 at 10 27 52](https://user-images.githubusercontent.com/145676/34249404-eeedfb32-e639-11e7-8eb7-a7b18be8e3a9.png)

🎉 Well done, team WebKit!

**Comment by wanderview** [comment]
> WPT results don't look too bright anywhere else though... (and also in Chromium, they're not perfect)

Looks better with the experimental branches:

https://wpt.fyi/results/fetch/api/request/destination?label=experimental

Firefox support will ship in stable 61 next week, I believe.

**Comment by nhoizey** [comment]
Hi everyone. I'm not seeking advice or any solution for my current issue, but my use case might help motivate (if needed) implementation of `request.destination`:

I've got the issue of resources loaded via a Service Worker that have a `.gif` extension, but actually a video content type.

The reason is Cloudinary's Fetch API allows me to convert an animated GIF to a video with parameters in the middle of the URL, but the URL's end is the source GIF file. Details in my recent blog post: [Using Cloudinary’s Fetch API to convert an animated GIF to a video](https://nicolas-hoizey.com/2018/08/using-cloudinary-s-fetch-api-to-convert-an-animated-gif-to-a-video.html).

`request.destination` is not ready everywhere yet, so I wonder how I should implement my caching strategy to cache actual GIFs but not these "strange" GIFs that are actually videos.

I used to filter URLs with the images extensions (because of the generic `*/*` Accept header), but now I have to accept `.gif` while removing videos.

I added a test for the presence of `/(f_mp4|f_webm)/` in the URL (that's the Cloudinary way to transform into another *f*ormat), but this isn't very satisfactory.

`request.destination` would be perfect to solve my issue.
